platforms/gke/base/core/teardown.sh

```diff
@@ -28,42 +32,13 @@ cd ${ACP_PLATFORM_CORE_DIR}/initialize &&
     terraform apply -input=false tfplan || exit 1
 rm tfplan
 
-cd ${ACP_PLATFORM_CORE_DIR}/workloads/kueue &&
-    echo "Current directory: $(pwd)" &&
-    terraform init &&
-    terraform destroy -auto-approve || exit 1
-rm -rf .terraform/ .terraform.lock.hcl
-
-# b/376312292
-# cd ${ACP_PLATFORM_CORE_DIR}/gke_enterprise/servicemesh &&
-#     echo "Current directory: $(pwd)" &&
-#     terraform init &&
-#     terraform destroy -auto-approve || exit 1
-# rm -rf .terraform/ .terraform.lock.hcl
-
-cd ${ACP_PLATFORM_CORE_DIR}/gke_enterprise/fleet_membership &&
-    echo "Current directory: $(pwd)" &&
-    terraform init &&
-    terraform destroy -auto-approve || exit 1
-rm -rf .terraform/ .terraform.lock.hcl
-
-cd ${ACP_PLATFORM_CORE_DIR}/container_node_pool &&
-    echo "Current directory: $(pwd)" &&
-    terraform init &&
-    terraform destroy -auto-approve || exit 1
-rm -rf .terraform/ .terraform.lock.hcl
-
-cd ${ACP_PLATFORM_CORE_DIR}/container_cluster &&
-    echo "Current directory: $(pwd)" &&
-    terraform init &&
-    terraform destroy -auto-approve || exit 1
-rm -rf .terraform/ .terraform.lock.hcl
-
-cd ${ACP_PLATFORM_CORE_DIR}/networking &&
-    echo "Current directory: $(pwd)" &&
-    terraform init &&
-    terraform destroy -auto-approve || exit 1
-rm -rf .terraform/ .terraform.lock.hcl
+for terraservice in "${terraservices[@]}"; do
+    cd "${ACP_PLATFORM_CORE_DIR}/${terraservice}" &&
+        echo "Current directory: $(pwd)" &&
+        terraform init &&
+        terraform destroy -auto-approve || exit 1
+    rm -rf .terraform/ .terraform.lock.hcl
```

Comment:
Shall we commit the dependency lock file instead of deleting it? This will make the environment more reproducible

https://developer.hashicorp.com/terraform/language/files/dependency-lock

Reply:
It should be committed in the repository, in my opinion, and not deleted here and other teardown scripts.

Not only it would help with dependency management because we could loosen version constraints, but it also provides checksumming for providers and modules.

Reply:
I will update the providers and generate the new lock files in my next PR.

Reply:
https://github.com/GoogleCloudPlatform/accelerated-platforms/pull/52